source/math/Quaternion.ooc

```diff
@@ -210,8 +224,51 @@ Quaternion: cover {
 		This new(realResult, imaginaryResult)
 	}
 	operator * (value: FloatPoint3D) -> FloatPoint3D {
-		(this * Quaternion new (0.0f, value) * this inverse) imaginary
+		(this * This new (0.0f, value) * this inverse) imaginary
+	}
+	operator [] (index: Int) -> Float {
+		result: Float
+		match(index) {
+			case 0 => result = this w
+			case 1 => result = this x
+			case 2 => result = this y
+			case 3 => result = this z
+			case => raise("Invalid index: #{index}, valid indices are 0-3.")
+		}
+		result
+	}
+	operator as -> String { this toString() }
+	toArray: func -> Float[] {
+		result := [this w, this x, this y, this z]
+		result
```

Comment:
I'm pretty sure this was fixed in https://github.com/fasterthanlime/rock/commit/1b88d707e8959bcba9da34a8cba5267f4b0e5f7b, but our version of rock doesn't have that fix.